core/food_apis/scheduler.py

```
@@ -116,6 +132,20 @@ async def stop(self):
+    async def wait_for_shutdown(self) -> None:
+        """Wait for a previously initiated async shutdown to complete.
+
+        RU: Дождаться завершения ранее запущенной остановки через сигнал/таск.
+        EN: If shutdown was triggered via signal handler (creating `_stop_task`),
+        await that task to ensure cleanup finishes before process exit.
+        """
+        task = self._stop_task
+        if task and not task.done():
+            try:
+                await task
+            except Exception as exc:  # pragma: no cover (best-effort logging)
```

Ruff (0.14.1)

Do not catch blind exception: `Exception`

(BLE001)